forest/cli/src/cli/attach_cmd.rs

```diff
@@ -0,0 +1,344 @@
+
+        // Add custom object that mimics `module.exports`
+        let moduleobj = JsObject::default();
+        moduleobj
+            .set("exports", JsValue::from(" "), false, context)
+            .unwrap();
+        context.register_global_property("module", JsValue::from(moduleobj), Attribute::default());
+
+        // Chain API
+        bind_func!(context, token, chain_get_name);
+
+        // Net API
+        bind_func!(context, token, net_addrs_listen);
+        bind_func!(context, token, net_peers);
+        bind_func!(context, token, net_disconnect);
+        bind_func!(context, token, net_connect);
+
+        // Sync API
+        bind_func!(context, token, sync_check_bad);
+        bind_func!(context, token, sync_mark_bad);
+        bind_func!(context, token, sync_status);
+
+        // Wallet API
+        // TODO: bind wallet_sign, wallet_verify
+        bind_func!(context, token, wallet_new);
+        bind_func!(context, token, wallet_default_address);
+        bind_func!(context, token, wallet_balance);
+        bind_func!(context, token, wallet_export);
+        bind_func!(context, token, wallet_import);
+        bind_func!(context, token, wallet_list);
+        bind_func!(context, token, wallet_has);
+        bind_func!(context, token, wallet_set_default);
+
+        // Message Pool API
+        bind_func!(context, token, mpool_push_message);
+
+        // Bind send_message
+        bind_func!(context, token, send_message);
+    }
+
+    fn import_prelude(&self, context: &mut Context) -> anyhow::Result<()> {
+        const INIT: &str = r"
+            const prelude = require('prelude.js')
+            prelude.greet();
+            if (prelude.showPeers) { showPeers = prelude.showPeers; }
+            if (prelude.getPeer) { getPeer = prelude.getPeer; }
+            if (prelude.disconnectPeers) { disconnectPeers = prelude.disconnectPeers; }
+            if (prelude.isPeerConnected) { isPeerConnected = prelude.isPeerConnected; }
+            if (prelude.showWallet) { showWallet = prelude.showWallet; }
+            if (prelude.showSyncStatus) { showSyncStatus = prelude.showSyncStatus; }
+            if (prelude.sendFIL) { sendFIL = prelude.sendFIL; }
+        ";
+        let result = context.eval(INIT);
+        if let Err(err) = result {
+            return Err(anyhow::anyhow!("error {err:?}"));
+        }
+
+        Ok(())
+    }
+
+    pub fn run(&self, config: Config) -> anyhow::Result<()> {
+        let mut context = Context::default();
+        self.setup_context(&mut context, &config.client.rpc_token);
+
+        // If only a short execution was requested, evaluate and return
+        if let Some(code) = &self.exec {
+            match context.eval(code.trim_end()) {
+                Ok(v) => match v {
+                    JsValue::Undefined => (),
+                    _ => println!("{}", v.display()),
+                },
+                Err(v) => eprintln!("Uncaught: {v:?}"),
```

Comment:
I guess this is giving lovecraftian errors

Reply:
Exactly.